### DISCUSSION TITLE
Update Mutation Duplicate key

### DISCUSSION THREAD

**Discussion by ibadass** [Akryum/vue-apollo#476] [Q&A]
Hi,

Could you help me for this problem with the update query (view-apollo / viewjs):
Here is my update method:
`updateCompany (id, name) {`
`	this.$apollo.mutate({`
`		mutation: UPDATE_COMPANY,`
`		variables: {`
`			username,`
`			firstname`
`		}`
`		update: (cache, {data: {updateCompany}}) => {`
`				const companies = cache.readQuery ({query: GET_COMPANIES})`
`				companies.companies.push (updateCompany)`
`				cache.writeQuery ({`
`				query: GET_COMPANIES,`
`				data: companies`
`			})`
`		}`
`	})`
`	.catch (error => {`
`		console.error (error)`
`	})`
`}`

Via my UI, I can modify the data, but I get this error:

> [Warn view]: Duplicate keys have been detected: '5c1ceca8348e0a0cfd82e'. This can cause an update error.

The data is edited, but the user interface displays the line twice with the same data, then I reload (manually) the page and the previously edited data is updated.

Thank you

**Comment by ydfzgyj** [comment]
You should not push the updated company to the `companies` list, but just edit the former data in cache.

Maybe you can format your code to help others read it.

**Comment by ibadass** [comment]
Thank you very much for your reply. I just delete the part about the cache and it's ok now.
